openff/units/openmm.py

```diff
@@ -191,3 +192,52 @@ def to_openmm_inner(quantity) -> "openmm_unit.Quantity":
         return to_openmm_inner(quantity)
     except MissingOpenMMUnitError:
         return to_openmm_inner(quantity.to_base_units())
+
+
+@requires_package("openmm.unit")
+def _ensure_openmm_quantity(
+    unknown_quantity: Union[Quantity, "openmm_unit.Quantity"]
+) -> "openmm_unit.Quantity":
+    if "openmm" in str(type(unknown_quantity)):
+        from openmm import unit as openmm_unit
+
+        if isinstance(unknown_quantity, openmm_unit.Quantity):
+            return unknown_quantity
+        else:
+            raise ValueError(
+                f"Failed to process input of type {type(unknown_quantity)}."
+            )
+    elif isinstance(unknown_quantity, Quantity):
+        return to_openmm(unknown_quantity)
+    else:
+        raise ValueError(f"Failed to process input of type {type(unknown_quantity)}.")
+
+
+def _ensure_openff_quantity(
+    unknown_quantity: Union[Quantity, "openmm_unit.Quantity"]
+) -> Quantity:
+    if isinstance(unknown_quantity, Quantity):
+        return unknown_quantity
+    elif "openmm" in str(type(unknown_quantity)):
+        from openmm import unit as openmm_unit
+
+        if isinstance(unknown_quantity, openmm_unit.Quantity):
+            return from_openmm(unknown_quantity)
+        else:
+            raise ValueError(
+                f"Failed to process input of type {type(unknown_quantity)}."
+            )
+    else:
+        raise Exception
+
+
+def ensure_quantity(
+    unknown_quantity: Union[Quantity, "openmm_unit.Quantity"],
+    type_to_ensure: Literal["openmm", "openff"],
+) -> Union[Quantity, "openmm_unit.Quantity"]:
+    if type_to_ensure == "openmm":
+        return _ensure_openmm_quantity(unknown_quantity)
+    elif type_to_ensure == "openff":
+        return _ensure_openff_quantity(unknown_quantity)
+    else:
+        raise Exception
```

Comment:
We could make the options an enum, it would make it more clear what options are allowed/helpw with a more descriptive error message.

Reply:
Agreed - just keeping it simple for now